### PR TITLE
chore: expose original principal from DefaultKsqlPrincipal (MINOR)

### PR DESCRIPTION
**Pull request by vcrfxia**
### Description 

Minor change to expose the original Principal wrapped inside DefaultKsqlPrincipal, so that custom extensions can access the original principal even after wrapping. This PR also updates the existing code to always wrap the original principal in DefaultKsqlPrincipal, and have DefaultKsqlPrincipal pass through the custom properties of the original principal, even if the original principal is a KsqlPrincipal. This change is made because otherwise we'd have two patterns for custom extensions to access original principals:
* have the original principal implement KsqlPrincipal (comes at no cost since KsqlPrincipal has a default implementation for the only method beyond the Principal interface that it extends), which causes the original principal to be passed through (throughout the ksql engine) directly, or
* do not implement KsqlPrincipal, which means the original principal will be wrapped in DefaultKsqlPrincipal. Then access the original principal from the new `getOriginalPrincipal()` method added in this PR.

To avoid confusion from having multiple patterns, this PR removes the first one by always wrapping the incoming Principal in a DefaultKsqlPrincipal.

This change needs to be backported to 7.1.x since that's where DefaultKsqlPrincipal was first introduced.

### Testing done 

Unit tests added.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-engine/src/main/java/io/confluent/ksql/security/DefaultKsqlPrincipal.java [hunk 1]
```diff
@@ -37,4 +37,8 @@ public String getName() {
   public Map<String, Object> getUserProperties() {
     return Collections.emptyMap();
   }
+
+  public Principal getOriginalPrincipal() {
```
Comment (spena) [inline ksqldb-engine/src/main/java/io/confluent/ksql/security/DefaultKsqlPrincipal.java:41]: Could you add a comment about where this method is used? Just to make sure is never deleted or public id removed accidentally.

**Review by spena — APPROVED** [review]

**Comment by spena** [comment]
@vcrfxia Seems we'll need to have only one pattern. Now that ksql wraps the JwtPrincipal, then custom plugins will never get a class with it, only KsqlPrincipal.

File: ksqldb-engine/src/main/java/io/confluent/ksql/security/DefaultKsqlPrincipal.java (hunk 1 above)
Comment (vcrfxia) [reply ksqldb-engine/src/main/java/io/confluent/ksql/security/DefaultKsqlPrincipal.java:54]: Done.

**Review by vcrfxia — COMMENTED** [review]
All right, I updated the code to always wrap the incoming Principal in a DefaultKsqlPrincipal, even if the incoming Principal is already a KsqlPrincipal. This cleans up the usage so there's only one access pattern for the original principal.

**Pull request merged**